File: transpose-dna.defunct.php
#! /usr/bin/env php
<?php

require_once ('include/php/proteomatic.php');

class TransposeDna extends ProteomaticScript
{
    function run()
    {
        // convert all characters to upper case
        $dna = strtoupper($this->param->nucleotides);
        // remove invalid characters
        $dna = preg_replace('/[^CGAT]/', '', $dna);
        // reverse sequence
        $dna = strrev($dna);
        // replace nucleotides
        $dna = strtr($dna, 'ACGT', 'TGCA');
        // output transposed DNA
        print($dna."\n");
        if (isset($this->output->result))
        {
            $f = fopen($this->output->result, 'w');
            fprintf($f, $dna."\n");
            fclose($f);
        }
    }
}

$script = new TransposeDna();
?>

File: transpose-dna.defunct.py
#! /usr/bin/env python

import sys, os
sys.path.append('./include/python')
import proteomatic
import string
import re

class TransposeDna(proteomatic.ProteomaticScript):
    def run(self):
        # convert all characters to upper case
        # Attention: parameters are Unicode because of the JSON parser 
        # used behind the scenes, convert nucleotides to ASCII string
        dna = str(self.param['nucleotides']).upper()
        # remove invalid characters
        dna = re.sub('[^ACGT]', '', dna)
        # reverse sequence
        dna = dna[::-1]
        # replace nucleotides
        dna = dna.translate(string.maketrans('ACGT', 'TGCA'))
        # output transposed DNA
        print(dna)
        
        if 'result' in self.output:
            with open(self.output['result'], 'w') as f:
                f.write(dna + "\n")
                
if __name__ == '__main__':
    script = TransposeDna()
